### PR TITLE
Enable, fix, and extend genrule stamping tests.

### PR DESCRIPTION
**Pull request by tmc**
The genrule stamping tests were not enabled and were broken.

Further, I extended them to cover the behavior of no explicit stamp attr value being provided.

### CODE REVIEW

**Comment by aiuto** [comment]
Guessed from CODEOWNERS.
Rule def in ./src/main/java/com/google/devtools/build/lib/rules/genrule/BUILD
CODEOWNERS => /src/main/java/com/google/devtools/build/lib/rules/ @lberki

Feel free to reassign.

**Comment by tmc** [comment]
@damienmg may also be interested.

File: src/test/shell/integration/stamping_test.sh [hunk 1]
```diff
@@ -29,32 +29,61 @@ function set_up() {
...
+genrule(
+  name = "unspecified",
+  outs = ["unspecified.txt"],
+  cmd = "(grep BUILD_TIMESTAMP bazel-out/volatile-status.txt || echo 'x 0') | cut -d ' ' -f 2 >\$@",
+)
 EOF
 }
 
+function expect_equals() {
+  [ "${1}" = "${2}" ]
```
Comment (lberki) [inline src/test/shell/integration/stamping_test.sh]: nit: use `assert_equals` to be consistent with the rest of our code base

**Comment by tmc** [comment]
@lberki rebased and pushed.

**Comment by lberki** [comment]
Thanks! I'll import this (and see if it breaks anything internally and if so, fix it)

**Comment by lberki** [comment]
(apologies for the delay)

/cc @meisterT 

I looked at this change a bit closer and I am now somewhat confused: if you set `stamp=0` on a `genrule`, it's not supposed to get `{stable,volatile}-status`.txt on its inputs. How does this test work then?

**Comment by meisterT** [comment]
Lukacs, you saw the `|| echo 'x 0'` in the genrule?

**Comment by lberki** [comment]
You're indeed right... but then why does this fail within Google with grep containing about non-existent files...? Let me explore a bit.

**Comment by lberki** [comment]
Well, that was embarrassing. I'll merge this.

**Pull request closed without merging**